test/state/page-config/actions.test.js

```diff
@@ -33,54 +42,150 @@ jest.mock('state/page-models/actions', () => ({
     .mockImplementation(require.requireActual('state/page-models/actions').setSelectedPageModel),
 }));
 
+jest.mock('state/page-models/helpers', () => ({
+  validatePageModel: jest.fn(),
+}));
+
 
 const resolve = arg => new Promise(r => r(arg));
 
 const mockStore = configureStore([thunk]);
 const INITIAL_STATE = rootReducer();
 const ERROR_MESSAGES = ERROR.errors.map(err => err.message);
 
+const CURRENT_PAGE_CODE = 'homepage';
+getParams.mockReturnValue({ pageCode: CURRENT_PAGE_CODE });
 
 describe('state/page-config/actions', () => {
-  beforeEach(jest.clearAllMocks);
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
 
   let store;
 
+  describe('setPageWidget()', () => {
+    const WIDGET_ID = 'widget_code';
+    const FRAME_ID = 1;
+    const OLD_FRAME_ID = 2;
+    let action;
+
+    beforeEach(() => {
+      action = setPageWidget(CURRENT_PAGE_CODE, WIDGET_ID, OLD_FRAME_ID, FRAME_ID);
+    });
+
+    it('returns an action with the correct action type', () => {
+      expect(action.type).toBe(SET_PAGE_WIDGET);
+    });
+
+    it('returns an action with a well formed payload', () => {
+      expect(action.payload).toEqual({
+        pageCode: CURRENT_PAGE_CODE,
+        widgetId: WIDGET_ID,
+        targetFrameId: FRAME_ID,
+        sourceFrameId: OLD_FRAME_ID,
+      });
+    });
+  });
+
+  describe('removePageWidgetSync()', () => {
+    const FRAME_ID = 1;
+    let action;
+
+    beforeEach(() => {
+      action = removePageWidgetSync(CURRENT_PAGE_CODE, FRAME_ID);
+    });
+
+    it('returns an action with the correct action type', () => {
+      expect(action.type).toBe(REMOVE_PAGE_WIDGET);
+    });
+
+    it('returns an action with a well formed payload', () => {
+      expect(action.payload).toEqual({
+        pageCode: CURRENT_PAGE_CODE,
+        frameId: FRAME_ID,
+      });
+    });
+  });
+
+
   describe('initConfigPage()', () => {
     beforeEach(() => {
+      fetchPage.mockReturnValue(resolve(HOMEPAGE_RESPONSE));
+      getPageModel.mockReturnValue(resolve(COMPLEX_RESPONSE));
+      validatePageModel.mockReturnValue([]);
       store = mockStore(INITIAL_STATE);
     });
 
     it('when GET /page/<code> returns errors, it will dispatch ADD_ERRORS', () => {
       fetchPage.mockReturnValue(resolve(ERROR));
-      store.dispatch(initConfigPage()).then(() => {
+      return store.dispatch(initConfigPage()).then(() => {
         const actionTypes = store.getActions().map(action => action.type);
         expect(actionTypes).toEqual([ADD_ERRORS]);
         expect(addErrors).toHaveBeenCalledWith(ERROR_MESSAGES);
       });
     });
 
     it('when GET /pagemodels/<code> returns errors, it will dispatch ADD_ERRORS', () => {
-      fetchPage.mockReturnValue(resolve(HOMEPAGE_RESPONSE));
       getPageModel.mockReturnValue(resolve(ERROR));
-      store.dispatch(initConfigPage()).then(() => {
+      return store.dispatch(initConfigPage()).then(() => {
```

Comment:
why the return?

Reply:
returning a Promise when testing asynchronous code, has the same effect than using done(), it syncronizes jest and the async code it's testing

Reply:
can we just use done since we are doing the same everywhere else?